### ISSUE TITLE
Possible crash when shadowing types and functions

### ISSUE THREAD

**Issue by arshajii** [seq-lang/seq#104]
It's possible to crash the compiler if we shadow types/functions like this:

```python
def foo(s: seq):  # using built-in seq type
    print s

foo(s'ACGT')

type seq(a: int)  # our new seq type shadows built-in one

def foo(s: seq):
    print s.a

foo(seq(42))
```

This is because the two `foo`s have the same mangled name, so the first one is actually re-used. Maybe disallow shadowing types and/or functions? Doing so will probably prevent a lot of bugs anyway.

**Comment by inumanag** [comment]
This will greatly reduce the pool of useful variable names as well... can we have smarter mangling in this case?

**Comment by inumanag** [comment]
Resolved in #132 .
